gi_wrapper_cpp/dicom_reader.cpp

```diff
@@ -119,35 +142,50 @@ double* DicomReader::acquireImage(DataSet* dataSet, int& dims,
             to_string(zSpacing) + ", " +
             to_string(zNumber);
     }
-    logWrapperMessage(parameterLog);
+    // logWrapperMessage(parameterLog);
 
     return array;
 }
 
-double DicomReader::acquireStart(DataSet* dataSet, int dim)
+
+double DicomReader::acquireStart(DataSet& dataSet, int dim)
 {
-    return dataSet->getDouble(TagId(tagId_t::ImagePositionPatient_0020_0032), dim);
+    Attribute<0x0020, 0x0032> imagePosition;
+    imagePosition.SetFromDataSet(dataSet);
+
+    return (double) imagePosition.GetValue(dim);
 }
 
-double DicomReader::acquireSpacing(DataSet* dataSet, int dim)
+double DicomReader::acquireSpacing(DataSet& dataSet, int dim)
 {
-    if (dim < 2)
+    if (dim < 2) {
+        Attribute<0x0028, 0x0030> pixelSpacing;
+        pixelSpacing.SetFromDataSet(dataSet);
         // We subtract dim from 1 because the order of pixel spacing values is inversed in DICOM standard.
         // More info - http://dicom.nema.org/medical/Dicom/2016b/output/chtml/part03/sect_10.7.html.
-        return dataSet->getDouble(TagId(tagId_t::PixelSpacing_0028_0030), 1 - dim);
+        return (double) pixelSpacing.GetValue(1 - dim);
+    }
+    
     else
     {
         double thirdDimSpacing = 0.0;
-        if (dataSet->getString(TagId(tagId_t::Modality_0008_0060), 0) == "RTDOSE" &&
-            dataSet->bufferExists(TagId(tagId_t::GridFrameOffsetVector_3004_000C), 0))
+        Attribute<0x0008, 0x0060> modality;
```

Comment:
@kdbogusz As said in other comment please extract those constants as some compile time constants (constexpr ...) to indicate the meaning of them